protocol/src/main/contraband/bsp.contra

```diff
@@ -132,6 +138,67 @@ type Diagnostic {
 
   ## The diagnostic's message.
   message: String!
+
+  ## A list of related diagnostic information, e.g. when symbol-names within
+  ## a scope collide all definitions can be marked via this property.
+  relatedInformation: [sbt.internal.bsp.DiagnosticRelatedInformation] @since("1.8.0")
+
+  ## Kind of data to expect in the `data` field. If this field is not set,
+  ## the kind of data is not specified.
+  dataKind: String @since("1.9.0")
+
+  ## A data entry field.
+  data: sbt.internal.bsp.ScalaDiagnostic @since("1.9.0")
+}
+
+## Represents a related message and source code location for a diagnostic.
+## This should be used to point to code locations that cause or are related to
+## a diagnostics, e.g when duplicating a symbol in a scope.
+type DiagnosticRelatedInformation {
+  ## The location of this related diagnostic information.
+  location: sbt.internal.bsp.Location!
+
+  ## The message of this related diagnostic information.
+  message: String!
+}
+
+## ScalaDiagnostic is a data structure that contains Scala-specific
+## metadata generated by Scala compilation. This metadata is
+## embedded in the `data: Option[Json]` field of the Diagnostic definition, when
+## the dataKind field contains "scala".
+type ScalaDiagnostic {
+  ## Actions (also known as quick fixes) that are able to either fix or address
+  ## the issue that is causing this diagnostic.
+  actions: [sbt.internal.bsp.ScalaAction]
+}
+
+## A Scala action represents a change that can be performed in code.
+## See also LSP: Code Action Request (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeAction).
+type ScalaAction {
```

Comment:
As pointed out by @ckipp01 in https://github.com/build-server-protocol/build-server-protocol/pull/527#pullrequestreview-1435850632 this is a duplication of `xsbti.Action`. I wonder if it would not be simpler to define the `xbsti.Action` directly as a `contra` instead. A counter argument is that maybe we don't want util-interface to depend on `sjsonnew`?

Reply:
util-interface and compiler-interface needs to be in Java, but I guess we can make a second subproject for sjsonnew codecs.

Reply:
Not sure if it would be better or worse though. I am fine with the PR as it is.